src/site/scripts/components/profile/actionbutton.tsx

```diff
@@ -88,25 +69,46 @@ export class ActionButton extends React.Component<IActionButtonProps, IActionBut
 
         className += ` action-${this.props.text.replace(/\s+/g, "-")}`;
 
+        if (this.props.confirmationText === undefined) {
+            return (
+            <div className={className}>
+                {this.renderButton(true)}
+            </div>);
+        } else {
         return (
             <div className={className}>
-                {this.renderButton()}
+                {this.renderButton(false)}
                 {this.renderConfirmation()}
             </div>);
+        }
     }
 
     /**
      * Renders the input button.
      * 
      * @returns The rendered input button.
      */
-    private renderButton(): JSX.Element {
-        return (
-            <input
-                className="action-button"
-                onClick={(): void => this.toggleExpansion()}
-                type="button"
-                value={this.props.text} />);
+    private renderButton(shortCircuitAction: boolean): JSX.Element {
+        if (shortCircuitAction) {
+            let onClick = (): void => {
```

Comment:
Even better: the only difference between the `input`s is the onClick. Instead of this big if/else, have one `const onClick` that's defined with a ternary.